pkg/pgmodel/migrate.go

```diff
@@ -457,14 +460,132 @@ func (t *Migrator) upgradeVersion(tx pgx.Tx, from, to semver.Version) error {
 	return nil
 }
 
-func installExtension(conn *pgxpool.Conn) {
-	_, extErr := conn.Exec(context.Background(), fmt.Sprintf(extensionInstall, extSchema))
-	if extErr != nil {
-		log.Warn("msg", "timescale_prometheus_extra extension not installed", "cause", extErr)
+func installExtension(conn *pgxpool.Conn) error {
+	availableVersions, err := getAvailableExtensionVersions(conn)
+	ExtensionIsInstalled = false
+	if err != nil {
+		return err
+	}
+	if len(availableVersions) == 0 {
+		log.Warn("msg", "timescale_prometheus_extra is not available, proceeding without extension")
+		return nil
+	}
+
+	newVersion, ok := getNewExtensionVersion(availableVersions)
+	if !ok {
+		log.Warn("msg", "timescale_prometheus_extra is not available at the right version, need version: %v, proceeding without extension", version.ExtVersionRangeString)
+		return nil
+	}
+
+	currentVersion, isInstalled, err := getInstalledExtensionVersion(conn)
+	if err != nil {
+		return fmt.Errorf("Could not get the installed extension version: %w", err)
+	}
+
+	if !isInstalled {
+		_, extErr := conn.Exec(context.Background(),
+			fmt.Sprintf("CREATE EXTENSION IF NOT EXISTS timescale_prometheus_extra WITH SCHEMA %s VERSION '%s'",
+				extSchema, getSqlVersion(newVersion)))
```

Comment:
One of these days we should decide on a policy for what gets inlined and what gets put in a `const` at the top. (My vote is everything that's used once should be inlined, it's easier to read, but once it's used twice it should be a `const` so it's easy to edit all usages)

Reply:
I agree with that policy